### ISSUE TITLE
Use sort -V for validate_data_dir.sh

### ISSUE THREAD

**Issue by iamanigeeit** [espnet/espnet#5264]
I could not get my `utt2spk` file to pass validation, and i traced it to strange `sort` [behaviour](https://stackoverflow.com/a/61139476/10092563) 

`example` in proper sorted order of `<speaker_id>_<utt_id>`:
```
1_2
1_4
13_1
```
```
$ cat example | sort
1_2
13_1
1_4
```
This sort order is crazy, and happens for some locales including `en_US.UTF-8`.

Unfortunately, using `LC_COLLATE=C` to enforce standard sorting does not help:
```
$ cat /tmp/utt2spk | LC_COLLATE=C sort
13_1
1_2
1_4
```
Now speaker_id 13 is before speaker_id 1, so `check_sorted_and_uniq $data/spk2utt` will fail. Only `sort -V` sorts correctly:

```
$ cat example | sort -V
1_2
1_4
13_1
```

**Comment by sw005320** [comment]
Thanks for the report.
So, would it be OK if we use the zero-padding as follows?
```
01_2
01_4
13_1
```
I think we do not encounter this issue so often because many examples use this way to align the IDs with the same character length.

**Comment by iamanigeeit** [comment]
> Thanks for the report. So, would it be OK if we use the zero-padding as follows?
> 
> ```
> 01_2
> 01_4
> 13_1
> ```
> 
> I think we do not encounter this issue so often because many examples use this way to align the IDs with the same character length.

Yes, that would also work for sorting. I'm not familiar with non-TTS corpora, so i don't know the implications of padding. Do we have to change the corpus dirnames and wav paths to match?

We could also relax the sorting requirements and just check for duplicates, because during training the audio gets sorted by length anyway.

**Comment by sw005320** [comment]
Oh, I see.
My example was for the utterance ID and speaker ID, which we can define.
Do you mean that it also happens in the paths?

**Comment by iamanigeeit** [comment]
> Do you mean that it also happens in the paths?

There's no requirement on the `*data_dir.sh` scripts on the wav paths as far as i can see -- `check_sorted_and_uniq` sorts on `utt_id` only. But i don't know if we have to zero-pad the corpus folders to match or not. For example, `LibriTTS/train-clean-100/26/495/` will need to be renamed to `LibriTTS/train-clean-100/0026/000495/`. If the wav filenames need to match utt_id, then they also need to be renamed.

**Comment by sw005320** [comment]
> If the wav filenames need to match utt_id, then they also need to be renamed.

No, `uttid` does not have to match the exact folder name.
